### PR TITLE
Add simplified ctor options for TableClient

### LINKED ISSUE #20522: [FEATURE REQ] TableClient constructor that takes a single Uri argument.
**Library or service name.**
Azure.Storage.Table
 
**Is your feature request related to a problem? Please describe.**
I'd like to request that we have a constructor to TableClient that takes a single SAS URL, in the same way that Microsoft.WindowsAzure.Storage.Table.CloudTable.cctor(Uri) did. You can do this because Table-level SAS's look like this:

https://STORAGEACCOUNT.table.core.windows.net/TABLENAME?si=ACCESSPOLICY&sv=2020-02-10&sig=sssssssssssssssssssssssssssssssssssssssssss%3D&tn=TABLENAME

So the TableClient can extract out all of the necessary information from the single URL that is given out when the SAS is created. Having a separate parameter for tableName and credentials is confusing when dealing with Table-level SASs.

### PR DESCRIPTION
**Pull request by christothes**
closes #20522

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 1]
```diff
@@ -54,23 +54,42 @@ public virtual string AccountName
...
+        /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline policies for authentication, retries, etc., that are applied to every request.
+        /// </param>
+        /// <exception cref="ArgumentException"><paramref name="endpoint"/> is not https.</exception>
+        public TableClient(Uri endpoint, TableClientOptions options = null)
+            : this(endpoint, null, default, options)
+        {
+            if (endpoint.Scheme != "https")
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:71]:
```suggestion
            if (endpoint.Scheme != Uri.UriSchemeHttps)
```

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 2]
```diff
@@ -54,23 +54,42 @@ public virtual string AccountName
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="TableClient"/> using the specified <see cref="Uri" /> <see cref="AzureSasCredential"/>.
+        /// Initializes a new instance of the <see cref="TableClient"/> using the specified <see cref="Uri" /> which contains a SAS token.
+        /// See <see cref="GetSasBuilder(TableSasPermissions, DateTimeOffset)" /> for creating a SAS token.
+        /// </summary>
+        /// <param name="endpoint">
+        /// A <see cref="Uri"/> referencing the table service account.
+        /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:62]: Would this have a different URL pattern due to the assumed embedding of the SAS and table name?

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 3]
```diff
@@ -54,23 +54,42 @@ public virtual string AccountName
...
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableClient"/> using the specified <see cref="Uri" /> and <see cref="AzureSasCredential"/>.
         /// See <see cref="GetSasBuilder(TableSasPermissions, DateTimeOffset)" /> for creating a SAS token.
         /// </summary>
         /// <param name="endpoint">
         /// A <see cref="Uri"/> referencing the table service account.
         /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:83]: Would this have a different URL pattern due to the assumed inclusion of the table name?

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 4]
```diff
@@ -54,23 +54,42 @@ public virtual string AccountName
...
         /// <exception cref="ArgumentException"><paramref name="endpoint"/> is not https.</exception>
-        public TableClient(Uri endpoint, string tableName, AzureSasCredential credential, TableClientOptions options = null)
-            : this(endpoint, tableName, default, credential, options)
+        public TableClient(Uri endpoint, AzureSasCredential credential, TableClientOptions options = null)
+            : this(endpoint, null, default, credential, options)
         {
-            Argument.AssertNotNull(tableName, nameof(tableName));
             Argument.AssertNotNull(credential, nameof(credential));
 
             if (endpoint.Scheme != "https")
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:95]:
```suggestion
            if (endpoint.Scheme != Uri.UriSchemeHttps)
```

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 5]
```diff
@@ -178,9 +195,24 @@ public TableClient(string connectionString, string tableName, TableClientOptions
 
         internal TableClient(Uri endpoint, string tableName, TableSharedKeyPipelinePolicy policy, AzureSasCredential sasCredential, TableClientOptions options)
         {
-            Argument.AssertNotNull(tableName, nameof(tableName));
             Argument.AssertNotNull(endpoint, nameof(endpoint));
 
+            // If we were provided no tableName, try to parse it from the endpoint.
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:200]: What if the endpoint and credential both specify a table name and they don't agree?

File: sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs [hunk 6]
```diff
@@ -135,6 +136,7 @@ public void ParsesSasStorage(string connString)
             Assert.That(GetCredString(tcs.Credentials), Is.EqualTo(SasToken), "The Credentials should have matched.");
             Assert.That(tcs.TableStorageUri.PrimaryUri, Is.EqualTo(new Uri($"https://{AccountName}.table.core.windows.net/?{SasToken}")), "The PrimaryUri should have matched.");
             Assert.That(tcs.TableStorageUri.SecondaryUri, Is.EqualTo(new Uri($"https://{AccountName}{TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix}.table.core.windows.net/?{SasToken}")), "The SecondaryUri should have matched.");
+            // Assert.That(tcs.);
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs:139]: nit:  This looks to be an accidental inclusion.

**Review by jsquire — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:62]: Good question - I'll update the example to reflect the additional information.

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs (hunk 5 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:200]: It will result in a 403

**Comment by christothes** [comment]
testing

**Pull request merged**